Backpack/Alert/Classes/BPKAlertView.m

```diff
@@ -0,0 +1,235 @@
+
+    [_contentHolderView addSubview:_titleLabel];
+    [_contentHolderView addSubview:_descriptionLabel];
+    [_contentHolderView addSubview:_buttonStackView];
+}
+
+- (void)setupConstraints {
+    [_circularView.topAnchor constraintEqualToAnchor:_circularView.superview.topAnchor].active = YES;
+    [_circularView.centerXAnchor constraintEqualToAnchor:_contentHolderView.centerXAnchor].active = YES;
+    [_circularView.heightAnchor constraintEqualToConstant:70].active = YES;
+    [_circularView.widthAnchor constraintEqualToConstant:70].active = YES;
+  
+    [_iconContainerView.centerYAnchor constraintEqualToAnchor:_circularView.centerYAnchor].active = YES;
+    [_iconContainerView.centerXAnchor constraintEqualToAnchor:_circularView.centerXAnchor].active = YES;
+    [_iconContainerView.heightAnchor constraintEqualToConstant:60].active = YES;
+    [_iconContainerView.widthAnchor constraintEqualToConstant:60].active = YES;
+   
+    [_iconImageView.centerYAnchor constraintEqualToAnchor:_iconContainerView.centerYAnchor].active = YES;
+    [_iconImageView.centerXAnchor constraintEqualToAnchor:_iconContainerView.centerXAnchor].active = YES;
+    
+    [_contentHolderView.topAnchor constraintEqualToAnchor:_circularView.bottomAnchor constant:-35].active = YES;
+    [_contentHolderView.bottomAnchor constraintEqualToAnchor:_contentHolderView.superview.bottomAnchor].active = YES;
+    [_contentHolderView.leadingAnchor constraintEqualToAnchor:_contentHolderView.superview.leadingAnchor].active = YES;
+    [_contentHolderView.trailingAnchor constraintEqualToAnchor:_contentHolderView.superview.trailingAnchor].active = YES;
+    
+    [_titleLabel.topAnchor constraintEqualToAnchor:_circularView.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_titleLabel.centerXAnchor constraintEqualToAnchor:_titleLabel.superview.centerXAnchor].active = YES;
+    
+    [_descriptionLabel.topAnchor constraintEqualToAnchor:_titleLabel.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_descriptionLabel.leadingAnchor constraintEqualToAnchor:_descriptionLabel.superview.leadingAnchor constant:BPKSpacingLg].active = YES;
+    [_descriptionLabel.trailingAnchor constraintEqualToAnchor:_descriptionLabel.superview.trailingAnchor constant:-BPKSpacingLg].active = YES;
+    
+    [_buttonStackView.topAnchor constraintEqualToAnchor:_descriptionLabel.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_buttonStackView.leadingAnchor constraintEqualToAnchor:_buttonStackView.superview.leadingAnchor constant:BPKSpacingXl].active = YES;
+    [_buttonStackView.trailingAnchor constraintEqualToAnchor:_buttonStackView.superview.trailingAnchor constant:-BPKSpacingXl].active = YES;
+    [_buttonStackView.bottomAnchor constraintEqualToAnchor:_buttonStackView.superview.bottomAnchor constant:-BPKSpacingLg].active = YES;
+}
+
+- (void)setupShadowOnView:(UIView *)view {
+    UIBezierPath *shadowPath = [UIBezierPath bezierPathWithRoundedRect:view.bounds cornerRadius:view.layer.cornerRadius];
```

Comment:
if you would like to be more granular (and design-right), you could add that circle to the path as well:
`UIBezierPath *path = [UIBezierPath bezierPathWith...];
    UIBezierPath *circle = [UIBezierPath bezierPathWithOvalInRect:...];
    [path appendPath:circle];`

Reply:
awesome, changed it from roundedRectangle and now I don't have to pass in the cornerradius.

Reply:
let me merge those paths to 1 and use BPKShadow